src/app/views/notifications/RouteNotifications.vue

```diff
@@ -0,0 +1,255 @@
+import { useGridStore } from '../../components/grid/grid-store';
+import { AppActivityFeedLazy } from '../../components/lazy';
+import AppShellNotificationPopoverStickerNavItem from '../../components/shell/notification-popover/sticker-nav-item/AppShellNotificationPopoverStickerNavItem.vue';
+import AppShellNotificationPopoverStickerNavItemPlaceholder from '../../components/shell/notification-popover/sticker-nav-item/AppShellNotificationPopoverStickerNavItemPlaceholder.vue';
+import { useAppStore } from '../../store';
+import { NotificationsFilterModal } from './filter/modal.service';
+import { routeNotifications } from './notifications.route';
+
+const HistoryCacheFeedTag = 'notifications-feed';
+
+const { unreadNotificationsCount, hasNewUnlockedStickers, markNotificationsAsRead } = useAppStore();
+
+const { grid } = useGridStore();
+
+const route = useRoute();
+
+const feed = ref<ActivityFeedView | null>(null);
+
+const itemsPerPage = ref(15);
+const totalStickersCount = ref(0);
+const isStickersLoading = ref(true);
+
+const routeTitle = computed(() => $gettext(`Your Notifications`));
+const shouldShowStickers = computed(() => totalStickersCount.value > 0);
+const shouldShowStickerPlaceholder = computed(() => isStickersLoading.value);
+
+const existingFilters = computed<string[] | undefined>(
+	() => feed.value?.extraData?.[NOTIFICATION_FILTER_FIELD]
+);
+
+const hasFilter = computed(
+	() =>
+		existingFilters.value &&
+		existingFilters.value.length !== SUPPORTED_NOTIFICATION_FEED_TYPES.length
+);
+
+watch(
+	unreadNotificationsCount,
+	() => {
+		if (
+			feed.value &&
+			!hasFilter.value &&
+			unreadNotificationsCount.value > feed.value.newCount
+		) {
+			feed.value.newCount = unreadNotificationsCount.value;
+		}
+	},
+	{ immediate: true }
+);
+
+createAppRoute({
+	routeTitle,
+	async onResolved({ payload, fromCache }) {
+		itemsPerPage.value = payload.perPage || itemsPerPage.value;
+
+		// TODO(notification-filtering) Scroll position isn't saved when clicking into a notification and navigating back.
+
+		feed.value = ActivityFeedService.routed(
+			null,
+			{
+				type: 'Notification',
+				name: 'notification',
+				url: `/web/dash/activity/more/notifications`,
+				extraData: {
+					[NOTIFICATION_FILTER_FIELD]: getNotificationTypesFromQuery(
+						route,
+						NOTIFICATION_FILTER_QUERY
+					),
+				},
+				itemsPerPage: itemsPerPage.value,
+			},
+			payload.items,
+			fromCache
+		);
+
+		// We mark in the history cache whether this route is a historical view
+		// or a new view. If it's new, we want to load fresh. If it's old, we
+		// want to use current feed data, just so we can try to go back to the
+		// correct scroll spot.
+		if (feed.value && !HistoryCache.has(route, HistoryCacheFeedTag)) {
+			feed.value.clear();
+			feed.value.append(Notification.populate(payload.items));
+			HistoryCache.store(route, true, HistoryCacheFeedTag);
+		}
+
+		if (!fromCache) {
+			grid.value?.pushViewNotifications('notifications');
```

Comment:
should we also only do this in the cases that they'd be able to "mark all as read"?